File: src/main/java/corp/soyummy/recipefavorites/RecipeFavoriteRepository.java
package corp.soyummy.recipefavorites;

import org.springframework.stereotype.Repository;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.bson.types.ObjectId;
import java.util.List;

@Repository
public interface RecipeFavoriteRepository extends MongoRepository<RecipeFavorite, String> {
    List<RecipeFavorite> findByRecipeId(ObjectId recipeId);
    List<RecipeFavorite> findByAmount(Integer amount);
    List<RecipeFavorite> findAllByOrderByAmountDesc();
}

File: src/main/kotlin/com/soyummy/recipefavorites/RecipeFavoriteRepository.kt
package com.soyummy.recipefavorites

import org.springframework.stereotype.Repository
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.bson.types.ObjectId

@Repository
interface RecipeFavoriteRepository : MongoRepository<RecipeFavorite, String> {
    fun findByRecipeId(recipeId: ObjectId): List<RecipeFavorite>
    fun findByAmount(amount: Int): List<RecipeFavorite>
    fun findAllByOrderByAmountDesc(pageable: Pageable): Page<RecipeFavorite>
}